backend/models/notification.js

```diff
@@ -0,0 +1,255 @@
+/**
+ *  Copyright (C) 2018 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+"use strict";
+const hasWriteAccessToModel = require("../middlewares/checkPermissions").hasWriteAccessToModelHelper;
+const job = require("../models/job");
+const utils = require("../utils");
+const uuid = require("node-uuid");
+const db = require("../db/db");
+const _ = require("lodash");
+
+const types = {
+	ISSUE_ASSIGNED : "ISSUE_ASSIGNED",
+	ISSUE_CREATED : "ISSUE_CREATED"
+};
+
+const NOTIFICATIONS_DB = "notifications";
+const MODELS_COLL = "settings";
+
+const generateNotification = function(type, data) {
+	const timestamp = (new Date()).getTime();
+	return Object.assign({_id:utils.stringToUUID(uuid.v1()), read:false, type, timestamp}, data);
+};
+
+const unionArrayMerger = function(objValue, srcValue) {
+	if (_.isArray(objValue)) {
+		return _.uniq(objValue.concat(srcValue));
+	}
+};
+
+/**
+ * Extract the teamspaceId/modelId info from an array of notifications
+ *
+ * @param {Notification[]} notifications The array of notifications which the data willl be extracted
+ * @returns {{keys..:Array<string>}} An object which keys are teamspaceId and an array of modelsIds as value
+ */
+const extractTeamSpaceInfo = function(notifications) {
+	return _.mapValues(_.groupBy(notifications, "teamSpace"), (notification) => _.map(notification, v => v.modelId));
+};
+
+/**
+ * Fills out the models name and extra data for the  modelids passed through parameter.
+ * @param {Object} teamSpaces an object which keys are teamspaces ids and values are an object witch keys are modelids
+ * @returns {Object} which contains the models data
+  */
+const getModelsData = function(teamSpaces) {
+	return Promise.all(
+		Object.keys(teamSpaces).map(ACCOUNT_DB => {
+			const modelsIds = teamSpaces[ACCOUNT_DB];
+
+			return db.getCollection(ACCOUNT_DB, MODELS_COLL)
+				.then(collection => collection.find({_id: {$in:modelsIds}}).toArray())
+				.then(models => {
+					const res = {};
+					const indexedModels = models.reduce((ac,c) => {
```

Comment:
merge this with line 72?
```suggestion
					res[ACCOUNT_DB]= models.reduce((ac,c) => {
```